File: server/python/app/infrastructure/mappers/payment_details/external_token_linked_mapper.py
from typing import Optional
from app.application.domain.payments.payment_details.external_token_linked import ExternalTokenLinked
from onlinepayments.sdk.domain.external_token_linked import ExternalTokenLinked as ExternalTokenLinkedSdk

class ExternalTokenLinkedMapper:

    @staticmethod
    def map_from_sdk_response(response: Optional[ExternalTokenLinkedSdk]) -> Optional[ExternalTokenLinked]:
        if response is None:
            return None

        dto = ExternalTokenLinked()
        dto.computed_token = response.computed_token
        dto.generated_token = response.generated_token
        dto.gts_computed_token = response.gts_computed_token

        return dto

File: server/ruby/app/infrastructure/mappers/payment_details/external_token_linked_mapper.rb
module Infrastructure
  module Mappers
    module PaymentDetails
      class ExternalTokenLinkedMapper
        def self.from_sdk_response(response)
          return nil if response.nil?

          dto = Business::Domain::Payments::PaymentDetails::ExternalTokenLinked.new
          dto.computed_token = response.computed_token
          dto.generated_token = response.generated_token
          dto.gts_computed_token = response.gts_computed_token

          dto
        end
      end
    end
  end
end
